Mage.Sets/src/mage/cards/i/IndominusRexAlpha.java

```diff
@@ -0,0 +1,183 @@
+package mage.cards.i;
+
+import java.util.*;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.AsEntersBattlefieldAbility;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.dynamicvalue.common.CountersSourceCount;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.cards.Card;
+import mage.cards.CardsImpl;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.counters.AbilityCounter;
+import mage.counters.Counter;
+import mage.counters.CounterType;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.target.common.TargetDiscard;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class IndominusRexAlpha extends CardImpl {
+
+    private static final DynamicValue xValue = new CountersSourceCount(null);
+
+    public IndominusRexAlpha(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{U/B}{U/B}{G}{G}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.DINOSAUR);
+        this.subtype.add(SubType.MUTANT);
+        this.power = new MageInt(6);
+        this.toughness = new MageInt(6);
+
+        // As Indominus Rex, Alpha enters the battlefield, discard any number of creature cards. It enters with a
+        // flying counter on it if a card discarded this way has flying. The same is true for first strike,
+        // double strike, deathtouch, hexproof, haste, indestructible, lifelink, menace, reach, trample, and vigilance.
+        this.addAbility(new AsEntersBattlefieldAbility(new IndominusRexAlphaCountersEffect()));
+
+        // When Indominus Rex enters the battlefield, draw a card for each counter on it.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new DrawCardSourceControllerEffect(xValue)));
+    }
+
+    private IndominusRexAlpha(final IndominusRexAlpha card) {
+        super(card);
+    }
+
+    @Override
+    public IndominusRexAlpha copy() {
+        return new IndominusRexAlpha(this);
+    }
+}
+
+// Based on MindMaggotsEffect
+class IndominusRexAlphaCountersEffect extends OneShotEffect {
+
+    private static final List<CounterType> copyableCounters = Arrays.asList(
+            CounterType.FLYING,
+            CounterType.FIRST_STRIKE,
+            CounterType.DOUBLE_STRIKE,
+            CounterType.DEATHTOUCH,
+            CounterType.HEXPROOF,
```

Comment:
There is more global problem with hexproof searching, see #12197. Possible fix for your card: search and collect abilities list and create ability counters with it (not predefined). Code from [[Wretched Bonemass]] can be useful.

Reply:
Kathril, Aspect Warper

Reply:
Reading that issue, it looks like the solution to this will be implemented as a fix to the abilities, not a fix to the cards that  care about this--would this better be handled by keeping the card the same and having the resulting PR fix the underlying inheritance asymmetry?

Reply:
Not for your card. It requires “counter list <-> abilities list” logic change too (it’s problem 2 from main issue).

But it’s ok here to keep old styled code with hexproof for main issue.

Current PR contains only one critical issue (merge stopper): need good actions order.

Reply:
Ah, so this needs to add the unconditional hexproof counter for abilities like "hexproof from blue"?

Reply:
Looks like we will need to implement counters for every hexproof condition in order to meet this: 
![image](https://github.com/magefree/mage/assets/17803304/c4ecbb72-17a1-4c34-a664-93a49a93d6bd)

I hesistate to start writing all those up though. is that the best way or can counters be made on the fly?

Reply:
Collect unique abilities and create counter manually by `new AbilityCounter` (not by `counterType.createInstance()`).

![IMG_0284](https://github.com/magefree/mage/assets/8344157/42e44a94-066e-4fdf-a400-e8031f030ccb)

Reply:
Refactored to make ability counters for every ability it copies as a counter. double check me, i think i have it right--all the tests still pass, and theyre checking for CounterType enums except for the testIndominusRexAlphaHexproofFromX test, which specifically throws a "hexproof from planeswalkers" ability into the mix.